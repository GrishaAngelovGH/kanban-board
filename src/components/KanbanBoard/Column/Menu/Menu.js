import ButtonGroup from "react-bootstrap/ButtonGroup"
import Dropdown from "react-bootstrap/Dropdown"
import DropdownButton from "react-bootstrap/DropdownButton"

const Menu = ({ markedAsDone, handleAddTask, handleMarkAsDone, handleDeleteAction, handleDeleteAllTasks, handleShowLimit, handleAutoAssignUsers }) => {
  const menuItems = [
    { label: "Add Task", icon: "bi bi-plus-circle-fill text-primary mx-1", onClick: handleAddTask },
    { label: markedAsDone ? "Unmark as Done" : "Mark as Done", icon: "bi bi-check-square-fill text-success mx-1", onClick: handleMarkAsDone },
    { label: "Delete Column", icon: "bi bi-x-circle-fill text-danger mx-1", onClick: handleDeleteAction },
    { label: "Delete All Tasks", icon: "bi bi-x-circle-fill text-danger mx-1", onClick: handleDeleteAllTasks },
    { label: "Set Task Limit", icon: "bi bi-exclamation-circle-fill text-warning mx-1", onClick: handleShowLimit },
    { label: "Auto Assign Users", icon: "bi bi-star-fill text-warning mx-1", onClick: handleAutoAssignUsers }
  ]

  return (
    <DropdownButton as={ButtonGroup} size="sm" variant="light" title={<i className="bi bi-three-dots-vertical"></i>}>
      {
        menuItems.map((v, i) => (
          <Dropdown.Item key={i} onClick={v.onClick}>
            <i className={v.icon}></i>
            {v.label}
          </Dropdown.Item>
        ))
      }
    </DropdownButton>
  )
}

export default Menu