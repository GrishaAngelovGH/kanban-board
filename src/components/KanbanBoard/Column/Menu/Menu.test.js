import { render, screen, fireEvent } from "@testing-library/react"

import Menu from "./Menu"

test("should render Column/Menu component", () => {
  const view = render(<Menu />)

  const toggleButton = screen.getByRole("button")

  fireEvent.click(toggleButton)

  expect(view).toMatchSnapshot()
})