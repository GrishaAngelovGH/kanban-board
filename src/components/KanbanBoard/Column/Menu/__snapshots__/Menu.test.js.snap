// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Column/Menu component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="show dropdown btn-group"
        role="group"
      >
        <button
          aria-expanded="true"
          class="dropdown-toggle show btn btn-light btn-sm"
          type="button"
        >
          <i
            class="bi bi-three-dots-vertical"
          />
        </button>
        <div
          aria-labelledby=""
          class="dropdown-menu show"
          x-placement="bottom-start"
        >
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-plus-circle-fill text-primary mx-1"
            />
            Add Task
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-check-square-fill text-success mx-1"
            />
            Mark as Done
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-x-circle-fill text-danger mx-1"
            />
            Delete Column
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-x-circle-fill text-danger mx-1"
            />
            Delete All Tasks
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-exclamation-circle-fill text-warning mx-1"
            />
            Set Task Limit
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            <i
              class="bi bi-star-fill text-warning mx-1"
            />
            Auto Assign Users
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="show dropdown btn-group"
      role="group"
    >
      <button
        aria-expanded="true"
        class="dropdown-toggle show btn btn-light btn-sm"
        type="button"
      >
        <i
          class="bi bi-three-dots-vertical"
        />
      </button>
      <div
        aria-labelledby=""
        class="dropdown-menu show"
        x-placement="bottom-start"
      >
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-plus-circle-fill text-primary mx-1"
          />
          Add Task
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-check-square-fill text-success mx-1"
          />
          Mark as Done
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-x-circle-fill text-danger mx-1"
          />
          Delete Column
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-x-circle-fill text-danger mx-1"
          />
          Delete All Tasks
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-exclamation-circle-fill text-warning mx-1"
          />
          Set Task Limit
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          <i
            class="bi bi-star-fill text-warning mx-1"
          />
          Auto Assign Users
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
